src/query/executor/engine.go

```diff
@@ -115,8 +115,6 @@ func (e *engine) Execute(
 	opts *QueryOptions,
 	results chan *storage.QueryResult,
 ) {
-	defer close(results)
```

Comment:
Hm, did you mean to delete this?

In the new code above you have the statement "// Results is closed by execute" which is not true anymore if this defer close doesn't occur?